### ISSUE TITLE
Creating wiggle files for every cell

### ISSUE THREAD

**Issue by AnniceNajafi** [YiPeng-Gao/scDaPars#22]
I am unsure of how I can make one Wiggle file for every cell. Should I split the bam file by barcode and then convert them to wig? Could you please help with this?

**Comment by Venteen123** [comment]
Did you solve it?

**Comment by Venteen123** [comment]
I have the same question

**Comment by AnniceNajafi** [comment]
Yes, I solved it but It's been a long time since I did that but I believe I used a function from the Sierra R package called SplitBam to generate multiple bam files. Here is the script I used for that:

```
library(Sierra)


# Load the barcodes file
barcodes_file <- "/scratch/user/annicenajafi/mani/cellranger_run_MCF10A_0Bd/outs/filtered_feature_bc_matrix/barcodes.tsv"
barcodes <- read.delim(barcodes_file, header = FALSE, stringsAsFactors = FALSE)
names(barcodes) <- "cellbc"

# Load the web summary file
web_summary_file <- "/scratch/user/annicenajafi/mani/cellranger_run_MCF10A_0Bd/outs/web_summary.html"
web_summary <- readLines(web_summary_file)

# Extract the number of reads assigned to cells and the number of cells detected
num_reads <- as.numeric(gsub("[^[:digit:]]", "", web_summary[71]))
num_cells <- as.numeric(gsub("[^[:digit:]]", "", web_summary[77]))

# Create the cell type column in the barcode data frame
cell_type <- rep("Cell", length.out = nrow(barcodes))

# Create the barcode data frame
barcode_df <- data.frame(celltype = cell_type, cellbc = barcodes$cellbc, stringsAsFactors = FALSE)



SplitBam(bam="/scratch/user/annicenajafi/mani/cellranger_run_MCF10A_0Bd/outs/possorted_genome_bam.bam",
         barcode_df, outdir="/scratch/user/annicenajafi/mani_split_bams/0Bd")
```

From there it should be easy to generate wiggle files for every cell. You can convert bam files to bed graphs then use the bed graphs to generate wigs. This forum may help:

https://www.biostars.org/p/2699/#14697


I hope this helps.

**Comment by Venteen123** [comment]
Dear Annice Najafi,

Thank you so much for taking the time to respond to my question regarding the scDaPars project and for sharing such a detailed solution!

The code example you provided using the SplitBamfunction from the Sierra package is extremely clear, particularly the complete workflow from reading cell barcodes from the barcodes.tsv file to generating the split BAM files. This will be tremendously helpful for my current single-cell data analysis work.

The forum link you recommended for BAM to wiggle file conversion is also very valuable. I will carefully study the various methods discussed there, especially the bedGraph to wig conversion process.

Your sharing has not only solved my current technical challenge but has also provided me with new approaches. Thank you again for your generous assistance!

Wishing you all the best with your research work. I look forward to potential future exchanges.

Sincerely,

wenting Jiang(江文婷)








C&nbsp;niu&nbsp;s
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "YiPeng-Gao/scDaPars"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年10月29日(星期三) 晚上10:23
***@***.***&gt;;
抄送:&nbsp;"C niu ***@***.******@***.***&gt;;
主题:&nbsp;Re: [YiPeng-Gao/scDaPars] Creating wiggle files for every cell (Issue #22)



AnniceNajafi left a comment (YiPeng-Gao/scDaPars#22)
 
Yes, I solved it but It's been a long time since I did that but I believe I used a function from the Sierra R package called SplitBam to generate multiple bam files. Here is the script I used for that:
 library(Sierra) # Load the barcodes file barcodes_file <- "/scratch/user/annicenajafi/mani/cellranger_run_MCF10A_0Bd/outs/filtered_feature_bc_matrix/barcodes.tsv" barcodes <- read.delim(barcodes_file, header = FALSE, stringsAsFactors = FALSE) names(barcodes) <- "cellbc" # Load the web summary file web_summary_file <- "/scratch/user/annicenajafi/mani/cellranger_run_MCF10A_0Bd/outs/web_summary.html" web_summary <- readLines(web_summary_file) # Extract the number of reads assigned to cells and the number of cells detected num_reads <- as.numeric(gsub("[^[:digit:]]", "", web_summary[71])) num_cells <- as.numeric(gsub("[^[:digit:]]", "", web_summary[77])) # Create the cell type column in the barcode data frame cell_type <- rep("Cell", length.out = nrow(barcodes)) # Create the barcode data frame barcode_df <- data.frame(celltype = cell_type, cellbc = barcodes$cellbc, stringsAsFactors = FALSE) SplitBam(bam="/scratch/user/annicenajafi/mani/cellranger_run_MCF10A_0Bd/outs/possorted_genome_bam.bam",          barcode_df, outdir="/scratch/user/annicenajafi/mani_split_bams/0Bd")  
From there it should be easy to generate wiggle files for every cell. You can convert bam files to bed graphs then use the bed graphs to generate wigs. This forum may help:
 
https://www.biostars.org/p/2699/#14697
 
I hope this helps.

**Comment by AnniceNajafi** [comment]
No problem! I am glad it helped.

Best,
Annice
